plugins/support_cc.lua

```diff
@@ -0,0 +1,54 @@
+-- mod-version:3 --lite-xl 2.1
+
+local core = require "core"
+local config = require "core.config"
+local common = require "core.common"
+
+local cc = common.merge({
+  backend = "gcc"
+}, config.plugins.support_cc)
+
+local function run_cmd(cmd, options)
+  local process = process.start(cmd, options)
+  local stdout, stderr = "", ""
+  while (process:running()) do
+    local stdout_chunk, stderr_chunk = process:read_stdout(), process:read_stderr()
+    if stdout_chunk ~= nil then stdout = stdout .. stdout_chunk end
+    if stderr_chunk ~= nil then stderr = stderr .. stderr_chunk end
+  end
+  local stdout_chunk, stderr_chunk = process:read_stdout(), process:read_stderr()
+  if stdout_chunk ~= nil then stdout = stdout .. stdout_chunk end
+  if stderr_chunk ~= nil then stderr = stderr .. stderr_chunk end
+  return process:returncode(), stdout, stderr
+end
+
+function cc.compile_plugin(target, options)
+  local dynamic_suffix = PLATFORM == "Mac OS X" and 'lib' or (PLATFORM == "Windows" and 'dll' or 'so')
```

Comment:
isn't it dylib on Mac OS?

Reply:
You're right, it is. Weird; I just copied this from start.lua; we'll want to change that there too.